inst/pages/MMUPHin_meta_analysis.qmd

```diff
@@ -0,0 +1,529 @@
+effect correction, and meta-analysis using the default random effects
+meta-regression.
+
+Here, we will first perform the batch effect correction analysis. We use
+both the relative abundance data and the pathway abundance data from 5
+public datasets: @Frankel2017, @Lee2022, @Matson2018, @Peters2019, and
+@Wind2020. All these datasets are available from the
+curatedMetagenomicData package, with a total of 285 subjects included.
+
+```{r load-pkg-data}
+
+######################
+# Load melanoma data #
+######################
+
+library(dplyr)
+
+se_relative <- sampleMetadata |>
+  filter(study_name %in% c("FrankelAE_2017", "GopalakrishnanV_2018", "LeeKA_2022", 
+                         "MatsonV_2018", "PetersBA_2019", "WindTT_2020")) |>
+  returnSamples("relative_abundance", rownames = "short")
+
+se_pathway <- sampleMetadata |>
+  filter(study_name %in% c("FrankelAE_2017", "GopalakrishnanV_2018", "LeeKA_2022", 
+                         "MatsonV_2018", "PetersBA_2019", "WindTT_2020")) |>
+  returnSamples("pathway_abundance", rownames = "short")
+
+```
+
+First, let's look into the relative abundance data.
+
+### Performing batch (study) effect adjustment with adjust_batch for relative abundance {#sec-batch-effect-adjustment}
+
+In this analysis, we are using two input files. The first input file is
+the relative abundance data file, which consists of a feature-by-sample
+matrix. The second input is the metadata associated with the relative
+abundance file, which includes the study names and overall response rate
+(ORR) the response variable along with other metadata. By the end of
+this step, we obtain a batch-adjusted relative abundance matrix.
+
+```{r prepare-input-data}
+
+library(MMUPHin)
+
+##########################
+# Create sample metadata #
+##########################
+data_meta <- select(as.data.frame(colData(se_relative)), c("study_name", "ORR", "PFS12", "anti_PD_1"))
+
+# Define response variable
+data_meta$resvar <- NA
+data_meta$anti_PD_1[data_meta$anti_PD_1 == "responder"] <- "yes"
+data_meta$anti_PD_1[data_meta$anti_PD_1 == "non_responder"] <- "no"
+data_meta$resvar[!is.na(data_meta$anti_PD_1)] <- data_meta$anti_PD_1[!is.na(data_meta$anti_PD_1)]
+data_meta$resvar[!is.na(data_meta$PFS12)] <- data_meta$PFS12[!is.na(data_meta$PFS12)]
+data_meta$resvar[!is.na(data_meta$ORR)] <- data_meta$ORR[!is.na(data_meta$ORR)]
+
+# Filter individuals without response variable
+data_meta <- data_meta[!is.na(data_meta$resvar), c("study_name", "resvar")]
+
+# Convert the "study_name" to factor variable
+data_meta$study_name <- as.factor(data_meta$study_name)
+
+###########################
+# Create Species Features #
+###########################
+# Transpose the abundance matrix and change the value of abundance data to proportion unit
+data_abd <- assay(se_relative)
+data_abd <- data_abd / 100
+
+# Match the individuals in the data_abd
+data_abd <- data_abd[, colnames(data_abd) %in% rownames(data_meta)]
```

Comment:
It might be a bit of work to achieve that given that we are in a multi-study setup and we are working on the metadata first to extract the individual studies and the associated changes that are independent of the assay data. Any workaround is appreciated.